android-kit-base/src/main/java/com/mparticle/kits/KitManagerImpl.java

```diff
@@ -1523,13 +1523,23 @@ private void confirmEmail(
         boolean hasHashedEmail = hashedEmail != null && !hashedEmail.isEmpty();
 
         if ((hasEmail || hasHashedEmail) && user != null) {
+            MParticle.IdentityType selectedIdentityType = null;
+            try {
+                String identityTypeStr = (kitConfiguration != null)
+                        ? kitConfiguration.getHashedEmailUserIdentityType()
+                        : null;
+                if (identityTypeStr != null && !identityTypeStr.equalsIgnoreCase("Unknown")) {
+                    selectedIdentityType = MParticle.IdentityType.valueOf(identityTypeStr);
+                }
+            } catch (IllegalArgumentException e) {
+                selectedIdentityType = MParticle.IdentityType.Other; // fallback if invalid value
```

Comment:
I don't think there should be a fallback to set this to a default. If there is no electedIdentityType, we shouldn't do any sort of identity request.